packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -102,14 +105,37 @@ function useHasTextColumnsControl( settings ) {
 
 function getUniqueFontSizesBySlug( settings ) {
 	const fontSizes = settings?.typography?.fontSizes;
-	const mergedFontSizes = fontSizes ? mergeOrigins( fontSizes ) : [];
-	const uniqueSizes = [];
-	for ( const currentSize of mergedFontSizes ) {
-		if ( ! uniqueSizes.some( ( { slug } ) => slug === currentSize.slug ) ) {
-			uniqueSizes.push( currentSize );
-		}
+	const mergedFontSizes = fontSizes
+		? mergeOrigins( fontSizes ).toReversed() // New array because mergeOrigins result is cached.
+		: [];
+	return uniqByProperty( mergedFontSizes, 'slug' ).reverse(); // In-place reverse to save memory.
+}
+
+/**
+ * Concatenate all the font sizes.
+ *
+ * @param {Object} settings The global styles settings.
+ *
+ * @return {Array} The merged font sizes.
+ */
+function getMergedFontSizes( settings ) {
+	/*
+	 * For backwards compatibility with presets converting from a hardcoded `false`
+	 * for `prevent_override` to a path to a boolean (`defaultFontSizes`, for example),
```

Comment:
`for `prevent_override` to a path to a boolean`

This could be rephrased. Also this comment should say why this is needed (as in what happens without it).